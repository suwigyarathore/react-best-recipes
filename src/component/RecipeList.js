const RecipeList = (props) => {
  return (
    <div style={props.style}>
      <h2 className="h2">RecipeList</h2>
      <ul className="list-reset">
        {props.recipes.map((recipe) => (
          <li
            className="py2 border-bottom border-bottom-dashed pointer"
            key={recipe.id}
            onClick={() => props.onClick(recipe.id)}
          >
            <span>{recipe.name}</span>
            <span>{recipe.category}</span>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default RecipeList;
