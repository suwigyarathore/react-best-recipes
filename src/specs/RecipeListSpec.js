import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../component/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Test Recipe',
    category: 'Test Category',
  },
  {
    id: 2,
    name: 'Test Recipe 2',
    category: 'Test Category 2',
  },
];

describe('<RecipeList/>', () => {
  test('Should not break when no recipes passed', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render when recipes passed', () => {
    const component = renderer.create(<RecipeList recipes={testRecipes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} favorites={[1]} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
