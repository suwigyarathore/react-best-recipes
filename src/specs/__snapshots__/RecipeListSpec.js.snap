// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList/> Should not break when no recipes passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList/> Should render favorite state correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mx1"
      onClick={[Function]}
      role="presentation"
    >
      ✅
    </span>
    <span>
      Test Recipe
    </span>
    <span>
      Test Category
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mx1"
      onClick={[Function]}
      role="presentation"
    >
      ⬜️
    </span>
    <span>
      Test Recipe 2
    </span>
    <span>
      Test Category 2
    </span>
  </li>
</ul>
`;

exports[`<RecipeList/> Should render when recipes passed 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mx1"
      onClick={[Function]}
      role="presentation"
    >
      ⬜️
    </span>
    <span>
      Test Recipe
    </span>
    <span>
      Test Category
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mx1"
      onClick={[Function]}
      role="presentation"
    >
      ⬜️
    </span>
    <span>
      Test Recipe 2
    </span>
    <span>
      Test Category 2
    </span>
  </li>
</ul>
`;
